### ISSUE TITLE
Recognise common URLs (when no exposed name or description provided)

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#10783]
### Is your feature request related to a problem? Please describe.

There are a number of "common" websites people link to with image buttons, and some widgets for these don't expose a name or description for these.  An example is just after the article heading on: https://www.pcworld.com/article/3519551/windows-10s-search-is-broken-here-are-some-workarounds.html there are links to share the article on Facebook, Twitter, Linked In, Reddit, by mail, or to print it respectively.

NVDA presents these as: "NVDA displays them as follows:

sharer u=https%3A%2F%2Fwww.pcw…  tweet url=https%3A%2F%2Fwww.pc…  shareArticle url=https%3A%2F%2…  submit url=https%3A%2F%2Fwww.p…  ?Subject=Update: Windows 10's search is broken. Here are some workarounds&Body=Check%20out%20this%20article%20from%20PCWorld%3A%20https%3A%2F%2Fwww.pcworld.com%2Farticle%2F3519551%2Fwindows-10s-search-is-broken-here-are-some-workarounds.html window.print();"

Other screen readers interpret these with the website at least, presumably based on the common URL.

### Describe the solution you'd like

Although this is clearly badly implemented by the site / widget, examples like this are common enough that it is worth considering a special case to interpret these.  I think this would be a fairly easy change to make, and make the web experience better for many. Done right it would very easy for add-ons to extend. One of the risks is it makes NVDA less useful for web testers, Reef proposed that we make this an optional feature.

**Comment by DrSooom** [comment]
Possible checkbox label "Decode HTML entities in URIs". I guess that this function should work in Browse as well as in Focus Mode the same way. Or did you mean something different, @Qchristensen?

**Comment by Qchristensen** [comment]
Yes that implementation sounds fine.  Maybe even a slightly simpler setting name (just thinking the users who would most benefit from this feature will probably also be able to locate it more easily with an easy to read setting name for it).

**Comment by DrSooom** [comment]
Should the decoding be limited to href, src, alt, longdesc, rel, title and onclick attributes only? I would prefer this limitation, otherwise the address bar and/or the clipboard content could be decoded as well. And I don't suppose that this behavior is desired.

And if we make this option enabled by default, I see no need to simplify the label in a more understandable way (for non-web developers).

**Comment by Qchristensen** [comment]
Sorry I just realised I never replied to your question.  Certainly my thought was around links on web pages, so yes I agree that limiting it to that context as you've proposed is a good idea.  It might also be worth identifying when that recognition has occurred so the user knows that.  Whether NVDA read "Likely Twitter link", or played an earcon for instance.
